src/engine/TransitivePathHashMap.cpp

```diff
@@ -46,8 +46,16 @@ HashMapWrapper TransitivePathHashMap::setupEdgesMap(
 }
 
 // _____________________________________________________________________________
-std::unique_ptr<Operation> TransitivePathHashMap::clone() const {
+std::unique_ptr<Operation> TransitivePathHashMap::cloneImpl() const {
   auto copy = std::make_unique<TransitivePathHashMap>(*this);
   copy->subtree_ = subtree_->clone();
+  if (copy->lhs_.treeAndCol_.has_value()) {
```

Comment:
Same here, maybe make this a member of the type of `lhs_ , then you can call it in both the implementations.